### DISCUSSION TITLE
Is there Token/Authorization mechanism there instead of basic auth?

### DISCUSSION THREAD

**Discussion by sunilsingh222** [exoplatform/exo-community#36] [Q&A]
Hi Team,

I wanted to use Token/Authorization mechanism instead of basic auth. I want to create token first time using basic auth and later based on token i wanted to use further APIs.

Thanks,
Sunil

**Comment by rdenarie** [comment]
Hello @sunilsingh222 

There is no token mechanism to access API in eXo Platform.

The only this you can do is to get the cookie JSESSIONID after successful authentication, and then reuse it to make API call. 
This cookie have a "short" lifetime, so it allow you to make few api call in a row, but you will not be able to reuse the cookie few days after for another call for example.

**Reply by sunilsingh222** [reply]
Hi @rdenarie ,

OAuth/JWT TOKEN needs to be externally by IDP like Keyckloak/Azure-AD, if no direct support available need inputs on how to achieve it else we may have to rethink using the eXo Platform.

**Reply by rdenarie** [reply]
Ok I think I did not understand your use case.

Can you explain a bit more what you try to do ?

In fact eXo Platform can be configured with some SSO systems like oauth, openid connect ...

Documentation for Oauth configuration for example : https://docs.exoplatform.org/administration/oauth-integration.html

**Reply by sunilsingh222** [reply]
Hi @rdenarie ,

Which options(SSO and openId) you have been provided all those are for UI flow and not for APIs flow. We need API based on  Oauth token which does work as rest API, so that we can utilise that token in further API i.e. create space, get space.

**Reply by rdenarie** [reply]
We only have UI flow for Oauth or openId. 

We don't have the API flow you need.

The only possibility you have is to simulate the ui flow with a sequence of requests to make the login, then extract the session cookie and use it for the API call.
